analyzer/src/main/kotlin/managers/Gradle.kt

```diff
@@ -134,6 +135,14 @@ class Gradle(
     // The path to the root project. In a single-project, just points to the project path.
     private lateinit var rootProjectDir: File
 
+    // Filter Gradle projects that are managed via Flutter / Pub. These projects are analyzed from within Pub.
+    override fun mapDefinitionFiles(definitionFiles: List<File>) =
+        definitionFiles.filter { gradleDefinitionFile ->
+            Pub.Factory().globsForDefinitionFiles.none { pubDefinitionFile ->
```

Comment:
The Pub Factory should not be instantiated for each definition file inside the `filter` expression; just create it once.

Usage of this factory creates a rather tight and cyclic coupling  between the `Pub` and the `Gradle` package manager implementations. It would be good if that cycle could be broken, so that only `Pub` knows `Gradle`, but not the other way around. But I do not know how this could be achieved with the current mechanism to identify definition files.

Reply:
I removed the Factory instantiation in the loop.